File: src/main/java/com/sleepwalker/sleeplib/util/MinecraftUtil.java
package com.sleepwalker.sleeplib.util;

import net.minecraft.util.ResourceLocation;
import net.minecraft.util.ResourceLocationException;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public final class MinecraftUtil {

    @Nullable
    public static ResourceLocation parseResourceLocationOrNull(@Nonnull String resourceLocation, @Nonnull String defNamespace){
        if(resourceLocation.indexOf(':') == -1){
            return ResourceLocation.tryCreate(defNamespace + ":" + resourceLocation);
        }
        else {
            return ResourceLocation.tryCreate(resourceLocation);
        }
    }

    @Nonnull
    public static ResourceLocation parseResourceLocation(@Nonnull String resourceLocation, @Nonnull String defNamespace) throws ResourceLocationException {
        if(resourceLocation.indexOf(':') == -1){
            return new ResourceLocation(defNamespace + ":" + resourceLocation);
        }
        else {
            return new ResourceLocation(resourceLocation);
        }
    }
}

File: src/main/kotlin/com/sleepwalker/sleeplib/util/MinecraftUtil.kt
package com.sleepwalker.sleeplib.util

import net.minecraft.util.ResourceLocation
import net.minecraft.util.ResourceLocationException

fun parseResourceLocationOrNull(resourceLocation: String, defNamespace: String): ResourceLocation? {
    return if (resourceLocation.indexOf(':') == -1) {
        ResourceLocation.tryCreate("$defNamespace:$resourceLocation")
    } else {
        ResourceLocation.tryCreate(resourceLocation)
    }
}

@Throws(ResourceLocationException::class)
fun parseResourceLocation(resourceLocation: String, defNamespace: String): ResourceLocation {
    return if (resourceLocation.indexOf(':') == -1) {
        ResourceLocation("$defNamespace:$resourceLocation")
    } else {
        ResourceLocation(resourceLocation)
    }
}